### PR TITLE
Fixes for OTP 18

### PR DESCRIPTION
**Pull request by fenek**
Used in https://github.com/esl/MongooseIM/pull/497, makes the library compatible with OTP 18.

### CODE REVIEW

**Comment by jonasrichard** [comment]
This pull request seems to be straightforward. What's about if some expert merges it? It is also blocking us to use riak erlang client on a project. Thanks

**Comment by koudelka** [comment]
Blocking Elixir 1.1 users, too. :(

**Comment by lukebakken** [comment]
I'll be running tests today to ensure we can merge this.

**Comment by koudelka** [comment]
<3

**Comment by lukebakken** [comment]
`make test` while using OTP 18.1 gives the following:

```
Generated src/riak_pb_messages.erl
==> riak_pb (compile)
Compiling src/riak_kv.proto

=INFO REPORT==== 5-Oct-2015::09:47:28 ===
Writing header file to "riak_kv_pb.hrl"
ERROR: compile failed while processing /home/lbakken/Projects/basho/riak-erlang-client/deps/riak_pb: {'EXIT',
    {{badmatch,
         {error,[],
             [{"src/riak_kv_pb.erl",
               [{244,erl_lint,{exported_var,'FNum',{'case',226}}}]}]}},
     [{protobuffs_compile,output,4,
          [{file,"src/protobuffs_compile.erl"},{line,147}]},
      {rebar_protobuffs_compiler,compile_each,2,
          [{file,"src/rebar_protobuffs_compiler.erl"},{line,120}]},
      {rebar_core,run_modules,4,[{file,"src/rebar_core.erl"},{line,446}]},
      {rebar_core,execute,5,[{file,"src/rebar_core.erl"},{line,371}]},
      {rebar_core,process_dir1,6,[{file,"src/rebar_core.erl"},{line,235}]},
      {rebar_core,process_each,5,[{file,"src/rebar_core.erl"},{line,305}]},
      {rebar_core,process_dir1,6,[{file,"src/rebar_core.erl"},{line,211}]},
      {rebar_core,process_commands,2,
          [{file,"src/rebar_core.erl"},{line,90}]}]}}
Makefile:6: recipe for target 'compile' failed
make: *** [compile] Error 1
```

Investigating.

**Comment by lukebakken** [comment]
This PR depends on basho/erlang_protobuffs#82

File: rebar.config [hunk 1]
```diff
@@ -1,8 +1,12 @@
-{require_otp_vsn, "R15|R16|17"}.
+{require_otp_vsn, "R15|R16|17|18"}.
 
 {cover_enabled, true}.
 {eunit_opts, [verbose]}.
-{erl_opts, [warnings_as_errors, debug_info, nowarn_deprecated_type]}.
```
Comment (lukebakken) [inline rebar.config]: Which warnings required removing `warnings_as_errors`? I've made a note of  basho/erlang_protobuffs#82 being a dependency of this PR.

Comment (fenek) [inline rebar.config]: OTP 18 produces warnings about deprecated `erlang:now/0` and investigating if it's safe to replace them was not in the scope of my work.

Comment (lukebakken) [inline rebar.config]: OK we will want to preserve the use of `warnings_as_errors` before merging this PR.

**Comment by jesuisbonbon** [comment]
<3

**Comment by fenek** [comment]
I've replaced commit that removes `warnings_as_errors` with a macro that will use alternative to `erlang:now/0` if OTP 18 is used. I've decided to do that to be on the safe side and avoid switching to `os:timestamp/0`.

**Comment by lukebakken** [comment]
Thank you @fenek I have incorporated your commits in PR #246

**Pull request closed without merging**